modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -238,6 +242,23 @@ public CommerceOrderItem addOrUpdateCommerceOrderItem(
 			CommerceContext commerceContext, ServiceContext serviceContext)
 		throws PortalException {
 
+		List<WorkflowTask> workflowTasks = _workflowTaskManager.search(
+			serviceContext.getCompanyId(), userId, null, null,
+			new String[] {CommerceOrder.class.getName()},
+			new Long[] {commerceOrderId}, null, null, null, null, false, null,
+			null, null, true, QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
+
+		for (WorkflowTask workflowTask : workflowTasks) {
+			String workflowTaskName = workflowTask.getName();
+
+			if (workflowTask.isAssignedToSingleUser() &&
+				StringUtil.equalsIgnoreCase(workflowTaskName, "review")) {
+
+				throw new CommerceWorkflowException(
+					"Order is already assigned");
+			}
+		}
```

Comment:
not sure, why don't you check the order status, I believe that should be enough. What do you think @luca-pellizzon ?

Reply:
We can't rely on just the order status, because we can have a buyer approval workflow that is started whe the order is still in "open" status.
If we check directly the workflow (if there already is a task) we are sure that a workflow is there.
I'm just not sure about the `workflowTask.isAssignedToSingleUser()` check. What if we have a task that is just not assigned yet? The order should be "blocked" anyway.